src/Module.Server/Rewards/CrpgRewardServer.cs

```diff
@@ -141,7 +141,9 @@ public override void OnRemoveBehavior()
         constantMultiplier = lowPopulationServer ? ExperienceMultiplierMin : constantMultiplier;
 
         CrpgRatingCalculator.UpdateRatings(_ratingResults);
-        Dictionary<PlayerId, PeriodStats> periodStats = _periodStatsHelper.ComputePeriodStats();
+        Dictionary<PlayerId, PeriodStats> periodStats = updateUserStats
+            ? _periodStatsHelper.ComputePeriodStats()
+            : new();

```

Comment:
I don't understand what's the issue. The first time this is called is at the end of the warmup, the result is not used since `updateUserStats` is true. Then it's called at the end of the first round and it computes the difference with the last time it was called, that is, end of warmup.

Here it seems like you are introducing the bug you are trying to fix. Since not calling that method for warmup would not reset the period.